tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/mapper/TaskMapper.java

```diff
@@ -43,7 +43,11 @@ public TaskSearchResponse toTaskSearchResponse(final TaskDTO taskDTO) {
             .setTaskState(taskDTO.getTaskState())
             .setSortValues(taskDTO.getSortValues())
             .setIsFirst(taskDTO.getIsFirst())
-            .setFormKey(taskDTO.getFormKey())
+            .setFormKey(
+                (taskDTO.getExternalFormReference() == null
+                        || taskDTO.getExternalFormReference().isBlank())
+                    ? taskDTO.getFormKey()
+                    : taskDTO.getExternalFormReference())
```

Comment:
the reason of not having a dedicated field for `externalFormReference` is to have something similar to JobWorkers ? @nathansandi